pkg/types/message_envelope_test.go

```diff
@@ -18,36 +19,179 @@ package types
+	assert.Equal(t, testCorrelationId, envelope.CorrelationID)
+	assert.Equal(t, ContentTypeJSON, envelope.ContentType)
+	assert.Equal(t, testPayload, string(envelope.Payload))
+	assert.Empty(t, envelope.QueryParams)
 }
 
 func TestNewMessageEnvelopeEmpty(t *testing.T) {
-
 	envelope := NewMessageEnvelope([]byte{}, context.Background())
 
+	assert.Equal(t, ApiVersion, envelope.ApiVersion)
+	assert.Empty(t, envelope.RequestID)
 	assert.Empty(t, envelope.CorrelationID)
 	assert.Empty(t, envelope.ContentType)
 	assert.Empty(t, envelope.Payload)
+	assert.Zero(t, envelope.ErrorCode)
+	assert.Empty(t, envelope.QueryParams)
+}
+
+func TestNewMessageEnvelopeForRequest(t *testing.T) {
+	expectedQueryParams := map[string]string{"foo": "bar"}
+	emptyQueryParams := map[string]string{}
+
+	tests := []struct {
+		name          string
+		queryParams   map[string]string
+		expectedEmpty bool
+	}{
+		{"valid - normal queryParams map", expectedQueryParams, false},
+		{"valid - empty queryParams map", emptyQueryParams, true},
+		{"valid - nil queryParams", nil, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			envelope := NewMessageEnvelopeForRequest([]byte(testPayload), tt.queryParams)
+
+			assert.NotEmpty(t, envelope.RequestID)
+			assert.NotEmpty(t, envelope.CorrelationID)
+			assert.Equal(t, ApiVersion, envelope.ApiVersion)
+			assert.Equal(t, testPayload, string(envelope.Payload))
+			assert.Equal(t, ContentTypeJSON, envelope.ContentType)
+			assert.Equal(t, 0, envelope.ErrorCode)
+			assert.NotNil(t, envelope.QueryParams)
+			if tt.expectedEmpty {
+				assert.Empty(t, envelope.QueryParams)
+			} else {
+				assert.Equal(t, expectedQueryParams, envelope.QueryParams)
+			}
+		})
+	}
+}
+
+func TestNewMessageEnvelopeForResponse(t *testing.T) {
+	invalidUUID := "123456"
+
+	tests := []struct {
+		name          string
+		correlationId string
+		requestId     string
+		contentType   string
+		expectedError bool
+	}{
+		{"valid", testCorrelationId, testRequestId, ContentTypeJSON, false},
+		{"invalid - CorrelationID is not in UUID format", invalidUUID, testRequestId, ContentTypeJSON, true},
+		{"invalid - invalid requestID is not in UUID format", testCorrelationId, invalidUUID, ContentTypeJSON, true},
+		{"invalid - ContentType is empty", testCorrelationId, testRequestId, "", true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			envelope, err := NewMessageEnvelopeForResponse([]byte(testPayload), tt.requestId, tt.correlationId, tt.contentType)
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.Equal(t, testRequestId, envelope.RequestID)
+				assert.Equal(t, testCorrelationId, envelope.CorrelationID)
+				assert.Equal(t, ApiVersion, envelope.ApiVersion)
+				assert.Equal(t, testPayload, string(envelope.Payload))
+				assert.Equal(t, ContentTypeJSON, envelope.ContentType)
+				assert.Equal(t, 0, envelope.ErrorCode)
+				assert.NotNil(t, envelope.QueryParams)
+			}
```

Comment:
Adding return avoids need to indent and makes code easier to follow. Also apply this to other tests.
```suggestion
			if tt.expectedError {
				assert.Error(t, err)
				return
			}
			
			assert.Equal(t, testRequestId, envelope.RequestID)
			assert.Equal(t, testCorrelationId, envelope.CorrelationID)
			assert.Equal(t, ApiVersion, envelope.ApiVersion)
			assert.Equal(t, testPayload, string(envelope.Payload))
			assert.Equal(t, ContentTypeJSON, envelope.ContentType)
			assert.Equal(t, 0, envelope.ErrorCode)
			assert.NotNil(t, envelope.QueryParams)
		
```